addons/l10n_in/wizard/l10n_in_withhold_wizard.py

```diff
@@ -138,6 +157,44 @@ def _compute_l10n_in_withholding_warning(self):
                 }
             wizard.l10n_in_withholding_warning = warnings
 
+    @api.depends('related_move_id', 'related_payment_id')
+    def _compute_tax_id(self):
+        for wizard in self:
+            sections = wizard.related_move_id._get_l10n_in_tds_tcs_applicable_sections()
+            if sections:
+                accounts_by_section = {}
+                for line in wizard.related_move_id.line_ids:
+                    section = line.account_id.l10n_in_tds_tcs_section_id
+                    if section in sections:
+                        accounts_by_section[section] = line.account_id
+                tax = self.env['account.tax']
+                for section, account in accounts_by_section.items():
+                    if not tax:
+                        # Search for the last withhold move line that matches the pan entity and account and section
+                        withhold_move_line = self.env['account.move.line'].search([
+                            ('move_id.l10n_in_withholding_ref_move_id.commercial_partner_id.l10n_in_pan_entity_id', '=', wizard.related_move_id.partner_id.commercial_partner_id.l10n_in_pan_entity_id.id),
+                            ('move_id.l10n_in_withholding_ref_move_id.line_ids.account_id', 'in', account.id),
```

Comment:
```suggestion
                            ('move_id.l10n_in_withholding_ref_move_id.line_ids.account_id.l10n_in_section_id', '=', section.id),
```
Does the account kind of determine the withholding tax more than the section?